### PR TITLE
Notifications

### PR DESCRIPTION
**Pull request by tom-sherman**
Todo:

- [x] DB migration
- [x] Notification page UI polish
- [x] Check DB query isn't totally horrendous for the amount of rows read
- [x] Make notification count consistent with notification page read state
- [x] Pagination

### CODE REVIEW

**Comment by tom-sherman** [comment]
I've checked the DB query and the rows read scales with the total number of notifications for a given user (including both read and unread). Not bad but ideally it would factor in the limit and/or read state.

I tried adding indexes on created_at and read_at, but that didn't help. Maybe I need to tweak the query a little, or maybe new indexes don't take into consideration old data????

**Comment by tom-sherman** [comment]
We agreed to mirror the bluesky UX as v1 which is to have no "mark as read" buttons and implicitly mark notifications as read when you open the page.

This means we can simply revalidate the notifications on navigation because the read state is much more ephemeral.

**Comment by tom-sherman** [comment]
@WillCorrigan this is a feature

**Comment by WillCorrigan** [comment]
@timothyis here are the initial cards, not much to look at and a bit same-y

all unread, dark mode: 
image

all unread, light mode: 
image

one marked as read (with the tick), dark mode:
image

one marked as read (with the tick), light mode: 
image

File: packages/frontpage/lib/schema.ts [hunk 1]
```diff
@@ -267,3 +265,12 @@ export const Report = sqliteTable("reports", {
     enum: ["pending", "accepted", "rejected"],
   }).default("pending"),
 });
+
+export const Notification = sqliteTable("notifications", {
+  id: integer("id").primaryKey(),
+  did: did("did").notNull(),
+  createdAt: dateIsoText("created_at").notNull().default(nowAsIsoString),
+  readAt: dateIsoText("read_at"),
```
Comment (WillCorrigan) [inline packages/frontpage/lib/schema.ts:273]: you can default this to strftime and leave it nullable maybe?

File: packages/frontpage/app/(app)/notifications/page.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+          <Notification key={notification.id} notification={notification} />
+        ))}
+      </>
+    ),
+
+    nextCursor: notifications.cursor,
+    pageSize: notifications.notifications.length,
+  };
+}
+// https://fall-ignored-ee-ga.trycloudflare.com/post/did:plc:6ieiiemxs4mvowwqjj5m4qxi/3l7xnhxbnvc2t/will.pds.dev.unravel.fyi/3l7xnhxbnvc2t
```
Comment (WillCorrigan) [inline packages/frontpage/app/(app)/notifications/page.tsx:74]:
```suggestion

```

File: packages/frontpage/drizzle/0004_illegal_boomerang.sql [hunk 3]
```diff
@@ -0,0 +1,9 @@
+CREATE TABLE `notifications`(
+  `id` integer PRIMARY KEY NOT NULL,
+  `did` text NOT NULL,
+  `created_at` text DEFAULT (strftime('%Y-%m-%dT%H:%M:%fZ', 'now')) NOT NULL,
+  `read_at` text,
```
[the remark this answers is not in the document]
Comment (tom-sherman) [reply packages/frontpage/drizzle/0004_illegal_boomerang.sql:5]: It's nullable so doesn't have a default.

File: packages/frontpage/lib/schema.ts (hunk 1 above)
Comment (tom-sherman) [reply packages/frontpage/lib/schema.ts:273]: Can't default it because it needs to be null to indicate not read.

**Pull request closed without merging**